inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -60,10 +60,53 @@ bool InLaneCruisingPlugin::onSpin()
   return true;
 }
 
+void InLaneCruisingPlugin::set_yield_client(ros::ServiceClient& client)
+{
+  yield_client_ = client;
+}
+
+bool InLaneCruisingPlugin::plan_trajectory_cb2(cav_srvs::PlanTrajectoryRequest& req,
+                                              cav_srvs::PlanTrajectoryResponse& resp)
+{
+
+  ROS_ERROR("in cb2");
+  if (yield_client_ && yield_client_.exists() && yield_client_.isValid()){
+    ROS_ERROR("Good Client");
+  }
+  cav_srvs::PlanTrajectory yield_srv;
+  if (req.initial_trajectory_plan.trajectory_id == "ILCReq"){
+    yield_srv.request.initial_trajectory_plan.trajectory_id = "YieldReq";
+  }
+
+  if(ros::service::exists("plugins/YieldPlugin/plan_trajectory", true))
+  {
+    ROS_ERROR("SERVICE EXISTS");
+
+    if(yield_client_.waitForExistence(ros::Duration(5.0))){
+      ROS_ERROR("SERVICE still EXISTS");
+    }
+    if (yield_client_.call(yield_srv))
+    {
+        ROS_ERROR("Yield SERVICE CALLED");
+        if (yield_srv.response.trajectory_plan.trajectory_id == "YieldResp")
+        {
+          resp.trajectory_plan.trajectory_id = "ILC2Yield";
+        } 
+    }
+    else
+      {
+        ROS_ERROR("Failed to call service ");
+      } 
+  }
+    else ROS_ERROR("Service Unavailable");
+  
+  return true;
+}
+
 bool InLaneCruisingPlugin::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req,
```

Comment:
Should I be expecting to see the yield plugin invoked here? Or am I missing where that happens?